cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataChangeRecord.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.metadata;
+
+import co.cask.cdap.proto.Id;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a Metadata change for a given {@link Id.NamespacedId}, including its previous state, the change,
+ * the time that the change occurred and (optionally) the entity that made the update.
+ */
+public final class MetadataChangeRecord {
+  private final Id.NamespacedId targetId;
+  private final MetadataRecord previous;
+  private final MetadataDiffRecord changes;
+  private final long updateTime;
+  private final String updater;
+
+  public MetadataChangeRecord(Id.NamespacedId targetId, MetadataRecord previous, MetadataDiffRecord changes,
+                              long updateTime) {
+    this(targetId, previous, changes, updateTime, null);
+  }
+
+  public MetadataChangeRecord(Id.NamespacedId targetId, MetadataRecord previous, MetadataDiffRecord changes,
+                              long updateTime, @Nullable String updater) {
+    this.previous = previous;
+    this.changes = changes;
+    this.targetId = targetId;
+    this.updateTime = updateTime;
+    this.updater = updater;
+  }
+
+  public Id.NamespacedId getTargetId() {
+    return targetId;
+  }
+
+  public MetadataRecord getPrevious() {
+    return previous;
+  }
+
+  public MetadataDiffRecord getChanges() {
+    return changes;
+  }
+
+  public long getUpdateTime() {
+    return updateTime;
+  }
+
+  @Nullable
+  public String getUpdater() {
+    return updater;
+  }
+
+  /**
+   * Represents the changes between the previous and the new record
+   */
+  public static final class MetadataDiffRecord {
+    private final MetadataRecord additions;
```

Comment:
Yes. I think we should leave it at delete + addition, primarily because tags do not have updates, only properties do.